internal/reconciler/logpipeline/otel/reconciler.go

```diff
@@ -43,14 +44,27 @@ type IstioStatusChecker interface {
 	IsIstioActive(ctx context.Context) bool
 }
 
+type AgentConfigBuilder interface {
+	Build(options agent.BuildOptions) *agent.Config
+}
+
+type AgentApplierDeleter interface {
+	ApplyResources(ctx context.Context, c client.Client, opts otelcollector.AgentApplyOptions) error
+	DeleteResources(ctx context.Context, c client.Client) error
+}
+
 var _ logpipeline.LogPipelineReconciler = &Reconciler{}
 
 type Reconciler struct {
 	client.Client
 
 	telemetryNamespace string
+	moduleVersion      string
```

Comment:
do we really need to make this configurable? it seems to me more like a constant or a build time argument at best